File: app/src/main/java/tv/hdonlinetv/besttvchannels/movies/watchfree/activity/ChannelViewModel.java
package tv.hdonlinetv.besttvchannels.movies.watchfree.activity;


import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import com.walhalla.data.model.Channel;

public class ChannelViewModel extends ViewModel {

    private final MutableLiveData<Channel> channelLiveData = new MutableLiveData<>();

    public LiveData<Channel> getChannel() {
        return channelLiveData;
    }

    public void setChannel(Channel channel) {
        channelLiveData.setValue(channel);
    }

    public void toggleFavorite() {
        Channel channel = channelLiveData.getValue();
        if (channel != null) {
            channel.liked = channel.liked > 0 ? 0 : 1;
            channelLiveData.setValue(channel);
        }
    }
}

File: app/src/main/java/tv/hdonlinetv/besttvchannels/movies/watchfree/activity/ChannelViewModel.kt
package tv.hdonlinetv.besttvchannels.movies.watchfree.activity

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import com.walhalla.data.model.Channel


class ChannelViewModel : ViewModel() {
    private val channelLiveData = MutableLiveData<Channel>()

    val channel: LiveData<Channel>
        get() = channelLiveData

    fun setChannel(channel: Channel) {
        channelLiveData.value = channel
    }

    fun toggleFavorite() {
        val channel = channelLiveData.value
        if (channel != null) {
            channel.liked = if (channel.liked > 0) 0 else 1
            channelLiveData.value = channel!!
        }
    }
}